### DISCUSSION TITLE
"<3" is not shown correctly in comments

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#20181] [Q&A] [answered]
Here are some random examples I found:

|  | [Map Link](https://osu.ppy.sh/beatmapsets/758986#fruits/1632603) ([comment](https://osu.ppy.sh/comments/1818384)) | [Map Link](https://osu.ppy.sh/beatmapsets/83494#osu/230523) ([comment](https://osu.ppy.sh/comments/236312))
|:---:|:---:|:---:|
| Lazer | ![image](https://user-images.githubusercontent.com/43073074/188965441-75ea509d-26d6-4438-b2ce-b8820a2ca2d1.png) | ![image](https://user-images.githubusercontent.com/43073074/188964773-577aee68-23f7-4aba-a41b-d528c8a71077.png)
| Web | ![image](https://user-images.githubusercontent.com/43073074/188965725-42cca9da-067f-466f-b799-1436b4b1f6e6.png) | ![image](https://user-images.githubusercontent.com/43073074/188964823-98ee039f-391e-486e-a5bc-1a678b140711.png)

(Hard to say whether it happens only with the characters "<" and "3")

**Comment by peppy** [comment]
Usually when characters show as `?` it means the original text is using non-standard unicode specification.

**Reply by smoogipoo** [reply]
Unless every tool is lying to me, it doesn't seem like the text is unicode.

**Reply by peppy** [reply]
```sql
root@db-master:osu> select * from comments where id = 1818384;
| id      | parent_id | user_id  | message | commentable_type | commentable_id | replies_count_cache | votes_count_cache | disqus_id | disqus_parent_id | disqus_thread_id | disqus_user_data | edited_at | edited_by_id | deleted_at | deleted_by_id | created_at          | updated_at          | pinned |
|---------|-----------|----------|---------|------------------|----------------|---------------------|-------------------|-----------|------------------|------------------|------------------|-----------|--------------|------------|---------------|---------------------|---------------------|--------|
| 1818384 | <null>    | 10777301 | <3<3    | beatmapset       | 758986         | 0                   | 0                 | <null>    | <null>           | <null>           | <null>           | <null>    | <null>       | <null>     | <null>        | 2021-09-03 18:00:03 | 2021-09-03 18:00:03 | 0      |
```

seems that way. not sure then.

**Comment by frenzibyte** [comment, marked as answer]
This is likely happening due to the emoji and safety extension used in markdown pipeline builder:
https://github.com/ppy/osu/blob/31a447fda0532fcf15f8c8251ca890c533386492/osu.Game/Graphics/Containers/Markdown/OsuMarkdownContainer.cs#L81-L85

**Reply by frenzibyte** [reply]
Will be resolved in https://github.com/ppy/osu/pull/20213
